### PR TITLE
Report orbit version in capture started event

### PR DESCRIPTION
**Pull request by dimitry-**
And warn users if capture is coming from the future.

Bug: http://b/189189746

### CODE REVIEW

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -338,6 +339,18 @@ void OrbitApp::OnCaptureStarted(const orbit_grpc_protos::CaptureStarted& capture
...
+        orbit_version::Version current_version = orbit_version::GetVersionNumber();
+        if (capture_version > current_version) {
+          main_window_->AppendToCaptureLog(
+              MainWindowInterface::CaptureLogSeverity::kWarning, absl::ZeroDuration(),
+              absl::StrFormat(
+                  "Capture appears to be coming from the future: it was captured by Orbit version "
+                  "%d.%d, please open the capture using Orbit v%d.%d or above",
+                  capture_version.major_version, capture_version.minor_version,
+                  capture_version.major_version, capture_version.minor_version));
+        }
```
Comment (dpallotti) [inline src/OrbitGl/App.cpp:353]: Should we also show a message box? This sounds like something the user should be made very well aware of.

File: src/OrbitGl/App.cpp [hunk 2]
```diff
@@ -338,6 +339,18 @@ void OrbitApp::OnCaptureStarted(const orbit_grpc_protos::CaptureStarted& capture
...
 
+        orbit_version::Version capture_version{capture_started.orbit_version_major(),
+                                               capture_started.orbit_version_minor()};
+        orbit_version::Version current_version = orbit_version::GetVersionNumber();
+        if (capture_version > current_version) {
+          main_window_->AppendToCaptureLog(
+              MainWindowInterface::CaptureLogSeverity::kWarning, absl::ZeroDuration(),
+              absl::StrFormat(
+                  "Capture appears to be coming from the future: it was captured by Orbit version "
+                  "%d.%d, please open the capture using Orbit v%d.%d or above",
```
Comment (dpallotti) [inline src/OrbitGl/App.cpp:350]: This string gives me a "cartoonish" feeling ("coming from the future"), suggestion for a different one:
"The capture was taken with Orbit version %d.%d, which is higher than the current version. Please open the capture using Orbit v%d.%d or above."
What do you think?
Also add a final period as all the other messages in the capture log have it.

Comment (pierricgimmig) [reply src/OrbitGl/App.cpp:350]: Yeah, I feel the same.

File: src/Service/CaptureServiceImpl.cpp [hunk 3]
```diff
@@ -227,8 +228,10 @@ static ProducerCaptureEvent CreateCaptureStartedEvent(const CaptureOptions& capt
     ERROR("%s", executable_path_or_error.error().message());
   }
 
-  capture_started->mutable_capture_options()->CopyFrom(capture_options);
   capture_started->set_capture_start_timestamp_ns(capture_start_timestamp_ns);
+  auto version = orbit_version::GetVersionNumber();
```
Comment (pierricgimmig) [inline src/Service/CaptureServiceImpl.cpp:232]: nit. auto -> orbit_version::Version? For simple types I really prefer not using auto. go/cstyle#Type_deduction

**Review by pierricgimmig — COMMENTED** [review]
Hi Dimitry, thanks for this, please see minor comments. Also, I think we should definitely warn the user with a dialog box.

File: src/OrbitGl/App.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp]: done.

File: src/OrbitGl/App.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp]: done.

File: src/OrbitGl/App.cpp [hunk 4]
```diff
@@ -338,6 +339,19 @@ void OrbitApp::OnCaptureStarted(const orbit_grpc_protos::CaptureStarted& capture
...
 
+        orbit_version::Version capture_version{capture_started.orbit_version_major(),
+                                               capture_started.orbit_version_minor()};
+        orbit_version::Version current_version = orbit_version::GetVersionNumber();
+        if (capture_version > current_version) {
+          std::string warning_message = absl::StrFormat(
+              "The capture was taken with Orbit version %d.%d, which is higher than the "
+              "current version. Please open the capture using Orbit v%d.%d or above.",
+              capture_version.major_version, capture_version.minor_version,
+              capture_version.major_version, capture_version.minor_version);
```
Comment (antonrohr) [inline src/OrbitGl/App.cpp:350]: Pers pref: use `absl::Substitute` instead of `StrFormat` here.

**Review by antonrohr — APPROVED** [review]

**Review by dpallotti — APPROVED** [review]

File: src/OrbitGl/App.cpp (hunk 4 above)
Comment (dimitry-) [reply src/OrbitGl/App.cpp]: done.

File: src/Service/CaptureServiceImpl.cpp (hunk 3 above)
Comment (dimitry-) [reply src/Service/CaptureServiceImpl.cpp]: done.

**Pull request merged**